packages/hoppscotch-common/src/newstore/environments.ts

```diff
@@ -37,14 +37,26 @@ type EnvironmentStore = typeof defaultEnvironmentsState
 
 const dispatchers = defineDispatchers({
   setSelectedEnvironmentIndex(
-    _: EnvironmentStore,
+    store: EnvironmentStore,
     {
       selectedEnvironmentIndex,
     }: { selectedEnvironmentIndex: SelectedEnvironmentIndex }
   ) {
-    return {
-      selectedEnvironmentIndex,
+    let isValidIndex = true
+
+    if (selectedEnvironmentIndex.type == "MY_ENV") {
+      isValidIndex =
+        selectedEnvironmentIndex.type == "MY_ENV" &&
```

Comment:
Won't this statement be always true because of the if condition above ?

Reply:
yes. i don't know what i was thinking while writing this.